tests/merch.test.ts

```diff
@@ -564,4 +569,81 @@ describe('merch order pickup events', () => {
     expect(persistedPickupEvent.orders).toHaveLength(1);
     expect(persistedPickupEvent.orders[0]).toStrictEqual(persistedOrder);
   });
+
+  test('placing an order with a pickup event that has reached the order limit fails', async () => {
+    const conn = await DatabaseConnection.get();
+    const member = UserFactory.fake({
+      points: 100,
+    });
+    const item = MerchFactory.fakeItem({
+      hidden: false,
+      monthlyLimit: 100,
+    });
+    const option = MerchFactory.fakeOption({
+      item,
+      quantity: 10,
+      price: 10,
+    });
+    const pickupEvent = MerchFactory.fakeOrderPickupEvent({
+      start: moment().add(3, 'days').toDate(),
+      end: moment().add(4, 'days').toDate(),
+      orderLimit: 2,
+    });
+
+    await new PortalState()
+      .createUsers(member)
+      .createMerchItem(item)
+      .createMerchItemOption(option)
+      .createOrderPickupEvents(pickupEvent)
+      .orderMerch(member, [{ option, quantity: 1 }], pickupEvent)
+      .orderMerch(member, [{ option, quantity: 1 }], pickupEvent)
+      .write();
+
+    const merchStoreController = ControllerFactory.merchStore(conn);
+    const placeMerchOrderRequest = {
+      order: [{ option: option.uuid, quantity: 1 }],
+      pickupEvent: pickupEvent.uuid,
+    };
+
+    await expect(merchStoreController.placeMerchOrder(placeMerchOrderRequest, member))
+      .rejects
+      .toThrow('Cannot place order with a fully-booked pickup event');
+  });
+
+  test('PATCH /order/pickup/:uuid fails if the order limit is decreased below the number of orders', async () => {
+    const conn = await DatabaseConnection.get();
+    const admin = UserFactory.fake({ accessType: UserAccessType.ADMIN });
+    const member = UserFactory.fake({
+      points: 100,
+    });
+    const item = MerchFactory.fakeItem({ hidden: false });
+    const option = MerchFactory.fakeOption({
+      item,
+      price: 10,
+    });
+    const pickupEvent = MerchFactory.fakeOrderPickupEvent({
+      start: moment().add(3, 'days').toDate(),
+      end: moment().add(4, 'days').toDate(),
+      orderLimit: 2,
+    });
```

Comment:
nit: there should be a factory method called `fakeFuturePickupEvent()` in the `OrderStates` branch that you can add to the factory so that you don't need to specify durations, since all we care about is that we can place orders on it:
```ts
  public static fakeFutureOrderPickupEvent(substitute?: Partial<OrderPickupEventModel>): OrderPickupEventModel {
    // guarantee that start and end are 2 days after current time so that orders can be placed
    const daysAhead = FactoryUtils.getRandomNumber(3, 10);
    const duration = FactoryUtils.getRandomNumber(30, 180, 30);
    const start = moment().add(daysAhead, 'days').toDate();
    const end = moment(start).add(duration, 'minutes').toDate();
    return MerchFactory.fakeOrderPickupEvent({ start, end, ...substitute });
  }
```

Reply:
Resolved in 075195b.